### PR TITLE
Feature/support subnet in config

### LINKED ISSUE #92: Support subnet in config
The monkey should be able to scan a specific subnet (or several subnets) by listing the subnet in the config, and not by specific IPs

### LINKED ISSUE #100: Rename "range" file to something that's not a python keyword
the file range is named as a python keyword. we should rename it

### PR DESCRIPTION
**Pull request by itaymmguardicore**
# Feature
> Closes #92 - Support configuring subnets for scanning.
> Closes #100 - Renamed range file

* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [X] Have you successfully tested your changes locally?

* Example screenshot/log transcript of the feature working
`2018-02-20 16:13:21,108 [19636:DEBUG] network_scanner.get_victim_machines.52: Scanning for potential victims in the network <FixedRange 192.168.1.34/30,192.168.2.50-192.168.2.54>`

##  Changes
  - fixed_range now supports subnets in formats: CIDR and ip ranges
  - Documentation of the config value updated

### CODE REVIEW

File: infection_monkey/network/info.py [hunk 1]
```diff
@@ -138,15 +139,14 @@ def get_ips_from_interfaces():
...
-        host_address = ipaddress.ip_address(address_str)
+        address_str = net_interface['addr']
+        netmask_str = net_interface['netmask']
         ip_interface = ipaddress.ip_interface(u"%s/%s" % (address_str, netmask_str))
         # limit subnet scans to class C only
         if ip_interface.network.num_addresses > 255:
-            ip_interface = ipaddress.ip_interface(u"%s/24" % address_str)
-        addrs = [str(addr) for addr in ip_interface.network.hosts() if addr != host_address]
-        res.extend(addrs)
+            res.append(CidrRange(cidr_range="%s/24" % (address_str, )))
```
Comment (danielguardicore) [inline infection_monkey/network/info.py]: Any reason we want to keep the class C limit?

File: common/network/range.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    @staticmethod
+    def _number_to_ip(num):
+        return socket.inet_ntoa(struct.pack(">L", num))
+
+
+class CidrRange(NetworkRange):
+    def __init__(self, cidr_range, shuffle=True):
+        super(CidrRange, self).__init__(shuffle=shuffle)
+        self._cidr_range = cidr_range.strip()
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply common/network/range.py]: Someone might explicitly instance CidrRange

File: common/network/range.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
+import random
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply common/network/range.py]: fixed

File: common/network/range.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    def __iter__(self):
+        base_range = self.get_range()
+        if self._shuffle:
+            random.shuffle(base_range)
+
+        for x in base_range:
+            yield self._number_to_ip(x)
+
+    @abstractmethod
+    def is_in_range(self, ip_address):
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply common/network/range.py]: Will be in use in detect-cross-segment-traffic

File: common/network/range.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
+import random
+import socket
+import struct
+from abc import ABCMeta, abstractmethod
+
+import ipaddress
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply common/network/range.py]: fixed

File: infection_monkey/network/info.py (hunk 1 above)
Comment (itaymmguardicore) [reply infection_monkey/network/info.py]: Was there previously. Logic says more than that is too much.
Your call whether to remove the limit.

File: common/network/range.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+
+class NetworkRange(object):
+    __metaclass__ = ABCMeta
+
+    def __init__(self, shuffle=True):
+        self._shuffle = shuffle
+
+    def get_range(self):
+        return [x for x in self._get_range() if (x & 0xFF != 0)]  # remove broadcast ips
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply common/network/range.py]: I'm removing this and adding a filter to CidrRange.

File: infection_monkey/network/info.py (hunk 1 above)
Comment (danielguardicore) [reply infection_monkey/network/info.py]: I'd rather remove these limits soon, or make it configurable. But  that depends on what we think is realistic. @ofriz ?

Comment (danielguardicore) [reply infection_monkey/network/info.py]: Fixed

**Pull request merged**